directories/src/redux/epics/node-epics.js

```diff
@@ -153,6 +161,47 @@ const requestGenesisHashEpic = (action$, store) => {
     });
 };
 
+const treasureHuntEpic = (action$, store) => {
+  return action$
+    .ofType(nodeActions.TREASURE_HUNT)
+    .mergeMap((action) => {
+      const { genesisHash, numberOfChunks } = action.payload;
+      const datamap = Datamap.generate(genesisHash, numberOfChunks);
+      const addresses = _.values(datamap);
+      const countSector = addresses / SECTOR_DIVIDER;
+      const randomSector = AppUtils.randomArray(1, countSector);
+      randomSector.map((sectorIndex) => {
+        let min = 0;
+        if((sectorIndex - 1) !== 0) {
+          min = (sectorIndex - 1) * SECTOR_DIVIDER;
+        }
+        const max = (sectorIndex * SECTOR_DIVIDER) - 1;
+        const randomSectorAddress = AppUtils.randomArray(min, max);
```

Comment:
Is this going to be picking a random address from within the sector?  We should be starting from the first address in the sector.  I.e. if you picked sector 1, start with the 1,000,000th address.

Reply:
newGenesis should start from 1st sector and random address
thats in description for oldGenesisHash.. no?